### PR TITLE
CM-113: add group to django orm

### PR DESCRIPTION
**Pull request by jdolkowski**
https://openimis.atlassian.net/browse/CM-113

It changes variable name used in social_protection module so it is required to also merge this: https://github.com/openimis/openimis-be-social_protection_py/pull/8

### CODE REVIEW

File: individual/schema.py [hunk 1]
```diff
@@ -44,18 +65,52 @@ def resolve_individual_data_source(self, info, **kwargs):
...
     update_individual = UpdateIndividualMutation.Field()
     delete_individual = DeleteIndividualMutation.Field()
+
+    create_group = CreateGroupMutation.Field()
+    update_group = UpdateGroupMutation.Field()
+    delete_group = DeleteGroupMutation.Field()
+
+    create_group_individual = CreateGroupIndividualMutation.Field()
+    update_group_individual = UpdateGroupIndividualMutation.Field()
+    delete_group_individual = DeleteGroupIndividualMutation.Field()
```
Comment (dborowiecki) [inline individual/schema.py:116]: I think those mutations could be named "add_individual_to_group", "edit_individual_in_group" and "remove_individual_from_group"

**Review by dborowiecki — CHANGES_REQUESTED** [review]
Code looks fine but I think we should add new mutation that will take Individuals uuid's as an input and then based on that it'll create groups (I believe that current solution requires us to at first create group and then call create_group_individual for all members, which requires multiple calls, "transactions" on frontend and is not very intuitive).

File: individual/schema.py (hunk 1 above)
Comment (jdolkowski) [reply individual/schema.py]: done

**Comment by jdolkowski** [comment]
The mutation has been added.

File: individual/services.py [hunk 2]
```diff
@@ -82,3 +86,34 @@ def update(self, obj_data):
...
+    def create(self, obj_data):
+        try:
+            with transaction.atomic():
+                individual_ids = obj_data.pop('individual_ids')
+                group = GroupService(self.user).create(obj_data)
+                group_id = group['data']['id']
+                individual_ids_list = [GroupIndividualService(self.user).create({'group_id': group_id,
+                                                                                 'individual_id': individual_id})
+                                       for individual_id in individual_ids]
+                group_and_individuals_dict = {**group, 'individuals': individual_ids_list}
```
Comment (dborowiecki) [inline individual/services.py:108]: Do we need this kind of output? Mixing group output with id's seems excessive. You can still access individuals through Group object. If you're worried about efficiency (like returning group here and fetching once again individuals later on), you can use select_related().

Also output should have format of `output_result_success`.

**Review by dborowiecki — CHANGES_REQUESTED** [review]

File: individual/services.py (hunk 2 above)
Comment (jdolkowski) [reply individual/services.py:108]: now it is in detail of **output_result_success**

File: individual/services.py [hunk 3]
```diff
@@ -82,3 +86,34 @@ def update(self, obj_data):
     @register_service_signal('group_individual.delete')
     def delete(self, obj_data):
         return super().delete(obj_data)
+
+
+class GroupFromMultipleIndividualsService(BaseService):
```
[the remark this answers is not in the document]
Comment (jdolkowski) [reply individual/services.py]: done

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**